sequencer/src/enriched_handle.rs

```diff
@@ -0,0 +1,55 @@
+use hotshot::{traits::NodeImplementation, types::SystemContextHandle};
+use hotshot_types::{light_client::StateSignature, traits::node_implementation::NodeType};
+use std::{
+    ops::{Deref, DerefMut},
+    sync::{Arc, RwLock},
+};
+
+use crate::api::signature_pool::StateSignaturePool;
```

Comment:
I wasn't convinced `signature_pool` belonged under the API module when I first saw it, and this convinces me. Conceptually, the API is like a leaf in the dependency graph of modules, so it's a bit weird to have a top-level module like this one importing things from the API module.

I would recommend making `src/state_signature/{signature_pool, enriched_handle}`, and then in the API we can import things from `crate::state_signature` as needed, rather than the other way around. It doesn't make any difference practically but I think it better captures the conceptual structure we're going for.

Reply:
Good idea. Will do